### PR TITLE
Fixed crashing on empty port range

### PR DESCRIPTION
**Pull request by vmanilo**
Github Issue: https://github.com/Twingate/terraform-provider-twingate/issues/77

## Changes
- fixed crashing when user passes empty string to ports range
- added tests

### CODE REVIEW

File: twingate/client_resource.go [hunk 1]
```diff
@@ -124,6 +128,12 @@ func convertPortsToSlice(a []interface{}) []string {
 	var res = make([]string, 0)
 
 	for _, elem := range a {
+		if elem == nil {
+			res = append(res, "")
+
+			continue
+		}
```
Comment (vmanilo) [inline twingate/client_resource.go:135]: @romankor here is the fix

File: twingate/resource_resource_test.go [hunk 2]
```diff
@@ -165,7 +165,124 @@ func testTwingateResource_fullFlowCreation(networkName, resourceName string) str
...
+	return fmt.Sprintf(`
+	resource "twingate_remote_network" "test" {
+	  name = "%s"
+	}
+
+	resource "twingate_resource" "test" {
+	  name = "%s"
+	  address = "updated-acc-test.com"
+	  remote_network_id = twingate_remote_network.test.id
+	  group_ids = ["R3JvdXA6MjMxNTQ="]
```
Comment (romankor) [inline twingate/resource_resource_test.go:201]: lets use `resource_group` to create this group and use it in the the test

File: twingate/resource_resource_test.go [hunk 3]
```diff
@@ -165,7 +165,124 @@ func testTwingateResource_fullFlowCreation(networkName, resourceName string) str
...
```
Comment (romankor) [inline twingate/resource_resource_test.go:272]: ttl

**Review by romankor — CHANGES_REQUESTED** [review]
See comments

File: twingate/resource_resource_test.go (hunk 2 above)
Comment (vmanilo) [reply twingate/resource_resource_test.go]: added resource_group

File: twingate/resource_resource_test.go (hunk 3 above)
Comment (vmanilo) [reply twingate/resource_resource_test.go]: added resource_group

**Review by romankor — APPROVED** [review]
LGTM

**Pull request merged**